test/api/v3/unit/models/challenge.test.js

```diff
@@ -0,0 +1,154 @@
+        let updatedNewMemberTasks = await Tasks.Task.find({_id: { $in: updatedNewMember.tasksOrder[`${taskType}s`]}});
+        let syncedTask = find(updatedNewMemberTasks, function findNewTask (updatedNewMemberTask) {
+          return updatedNewMemberTask.type === taskValue.type && updatedNewMemberTask.text === taskValue.text;
+        });
+
+        expect(updatedNewMember.challenges).to.contain(challenge._id);
+        expect(updatedNewMember.tags[3]._id).to.equal(challenge._id);
+        expect(updatedNewMember.tags[3].name).to.equal(challenge.shortName);
+        expect(syncedTask).to.exist;
+      });
+
+      it('updates tasks to challenge and challenge members', async () => {
+        let updatedTaskName = 'Updated Test Habit';
+        await challenge.addTasks([task]);
+
+        _.assign(task, _.merge(task.toObject(), Tasks.Task.sanitizeUpdate({ text: updatedTaskName })));
+        await challenge.updateTask(task);
+
+        let updatedLeader = await User.findOne({_id: leader._id});
+        let updatedUserTask = await Tasks.Task.findById(updatedLeader.tasksOrder[`${taskType}s`][0]);
+
+        expect(updatedUserTask.text).to.equal(updatedTaskName);
+      });
+
+      it('removes a tasks to challenge and challenge members', async () => {
+        await challenge.addTasks([task]);
+        await challenge.removeTask(task);
+
+        let updatedLeader = await User.findOne({_id: leader._id});
+        let updatedUserTask = await Tasks.Task.findOne({_id: updatedLeader.tasksOrder[`${taskType}s`][0]}).exec();
+
+        expect(updatedUserTask.challenge.broken).to.equal('TASK_DELETED');
+      });
+
+      it('unlinks and deletes challenge tasks for a user when remove-all is specified', async () => {
+        await challenge.addTasks([task]);
+        await challenge.unlinkTasks(leader, 'remove-all');
+
+        let updatedLeader = await User.findOne({_id: leader._id});
+        let updatedLeadersTasks = await Tasks.Task.find({_id: { $in: updatedLeader.tasksOrder[`${taskType}s`]}});
+        let syncedTask = find(updatedLeadersTasks, function findNewTask (updatedLeadersTask) {
+          return updatedLeadersTask.type === taskValue.type && updatedLeadersTask.text === taskValue.text;
+        });
+
+        expect(syncedTask).to.not.exist;
+      });
+
+      it('unlinks and keeps challenge tasks for a user when keep-all is specified', async () => {
+        await challenge.addTasks([task]);
+        await challenge.unlinkTasks(leader, 'keep-all');
+
+        let updatedLeader = await User.findOne({_id: leader._id});
+        let updatedLeadersTasks = await Tasks.Task.find({_id: { $in: updatedLeader.tasksOrder[`${taskType}s`]}});
+        let syncedTask = find(updatedLeadersTasks, function findNewTask (updatedLeadersTask) {
+          return updatedLeadersTask.type === taskValue.type && updatedLeadersTask.text === taskValue.text;
+        });
+
+        expect(syncedTask).to.exist;
+        // expect(syncedTask.challenge).to.be.empty;
```

Comment:
@TheHollidayInn It may be that the syncedTask object is a class and so has some hidden properties? Actually syncedTask.challenge is not a class... if you do `expect(syncedTask.challenge).to.eql({});` or Object.keys(syncedTask).length === 0